### DISCUSSION TITLE
(Info in case anyone wants to PR an update) Helix's tree-sitter-bash is 2 years out of date and missing many features

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#10476] [General]
Last update [was over 2 years ago](https://github.com/helix-editor/helix/pull/1917) and we are missing a lot of fixes and features:

 -   Replace non-JSON-standard single-quotes with double-quotes in binding.gyp by [@nordlow](https://github.com/nordlow) in [https://github.com/tree-sitter/tree-sitter-bash/pull/240](https://github.com/tree-sitter/tree-sitter-bash/pull/240)
 -   Fixes by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/245](https://github.com/tree-sitter/tree-sitter-bash/pull/245)

 -   fix: heredocs with many trailing file redirects by [@ahlinc](https://github.com/ahlinc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/197](https://github.com/tree-sitter/tree-sitter-bash/pull/197)
 -   fix: don't expose `\n` rule anchors as visible anonymous nodes by [@ahlinc](https://github.com/ahlinc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/198](https://github.com/tree-sitter/tree-sitter-bash/pull/198)
 -   fix: restore missed `redirect` field by [@ahlinc](https://github.com/ahlinc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/200](https://github.com/tree-sitter/tree-sitter-bash/pull/200)
 -   fix: alias `simple_heredoc_body` to `heredoc_body` by [@ahlinc](https://github.com/ahlinc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/201](https://github.com/tree-sitter/tree-sitter-bash/pull/201)

 -   Fixes and fixes by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/191](https://github.com/tree-sitter/tree-sitter-bash/pull/191)
 -   Fixes by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/192](https://github.com/tree-sitter/tree-sitter-bash/pull/192)
 -   Heredoc rework by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/193](https://github.com/tree-sitter/tree-sitter-bash/pull/193)

 -   Fix Cargo repository URL by [@Sighery](https://github.com/Sighery) in [https://github.com/tree-sitter/tree-sitter-bash/pull/97](https://github.com/tree-sitter/tree-sitter-bash/pull/97)
 -   fix: escaped newline immediately after a char, resolves [https://github.com/tree-sitter/tree-sitter-bash/issues/100](https://github.com/tree-sitter/tree-sitter-bash/issues/100) by [@ahlinc](https://github.com/ahlinc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/102](https://github.com/tree-sitter/tree-sitter-bash/pull/102)
 -   Fixed CRLF behavior for tests, run tests on all platforms in GitHub CI by [@ahelwer](https://github.com/ahelwer) in [https://github.com/tree-sitter/tree-sitter-bash/pull/106](https://github.com/tree-sitter/tree-sitter-bash/pull/106)
 -   Support for 'select' loops by [@mjambon](https://github.com/mjambon) in [https://github.com/tree-sitter/tree-sitter-bash/pull/111](https://github.com/tree-sitter/tree-sitter-bash/pull/111)
 -   Add support for 'until' loops by [@mjambon](https://github.com/mjambon) in [https://github.com/tree-sitter/tree-sitter-bash/pull/112](https://github.com/tree-sitter/tree-sitter-bash/pull/112)
 -   Handle words containing bare '#' by [@oxalica](https://github.com/oxalica) in [https://github.com/tree-sitter/tree-sitter-bash/pull/109](https://github.com/tree-sitter/tree-sitter-bash/pull/109)
 -   adding zsh expansion flags by [@ryaminal](https://github.com/ryaminal) in [https://github.com/tree-sitter/tree-sitter-bash/pull/115](https://github.com/tree-sitter/tree-sitter-bash/pull/115)
 -   Update CI by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/131](https://github.com/tree-sitter/tree-sitter-bash/pull/131)
 -   Update Cargo.toml by [@nokome](https://github.com/nokome) in [https://github.com/tree-sitter/tree-sitter-bash/pull/117](https://github.com/tree-sitter/tree-sitter-bash/pull/117)
 -   Rename ansii\_c\_string and string\_expansion by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/121](https://github.com/tree-sitter/tree-sitter-bash/pull/121)
 -   rust: enables highlights query by [@Dav1dde](https://github.com/Dav1dde) in [https://github.com/tree-sitter/tree-sitter-bash/pull/132](https://github.com/tree-sitter/tree-sitter-bash/pull/132)
 -   Swift Package Manager by [@lukepistrol](https://github.com/lukepistrol) in [https://github.com/tree-sitter/tree-sitter-bash/pull/124](https://github.com/tree-sitter/tree-sitter-bash/pull/124)
 -   Fix scanning of heredoc\_body to allow empty bodies by [@jaopaulolc](https://github.com/jaopaulolc) in [https://github.com/tree-sitter/tree-sitter-bash/pull/137](https://github.com/tree-sitter/tree-sitter-bash/pull/137)
 -   \[fix\] Here-documents: parse a “real” shell word (or close enough) after `<<` by [@domq](https://github.com/domq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/142](https://github.com/tree-sitter/tree-sitter-bash/pull/142)
 -   Parse Bash's tests by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/135](https://github.com/tree-sitter/tree-sitter-bash/pull/135)
 -   Fix CI by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/145](https://github.com/tree-sitter/tree-sitter-bash/pull/145)
 -   Support file descriptors for here docs/strings by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/156](https://github.com/tree-sitter/tree-sitter-bash/pull/156)
 -   Support optional opening paren in case by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/157](https://github.com/tree-sitter/tree-sitter-bash/pull/157)
 -   Highlight "select" and "until" as keywords by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/168](https://github.com/tree-sitter/tree-sitter-bash/pull/168)
 -   Undo misguided package.json changes by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/173](https://github.com/tree-sitter/tree-sitter-bash/pull/173)
 -   Restore prebuild dependencies by [@verhovsky](https://github.com/verhovsky) in [https://github.com/tree-sitter/tree-sitter-bash/pull/174](https://github.com/tree-sitter/tree-sitter-bash/pull/174)
 -   feat: rewrite the scanner in C by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/179](https://github.com/tree-sitter/tree-sitter-bash/pull/179)
 -   fix: make helper functions static to avoid compilation conflicts with other parsers by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/182](https://github.com/tree-sitter/tree-sitter-bash/pull/182)
 -   Fixes by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/186](https://github.com/tree-sitter/tree-sitter-bash/pull/186)
 -   fix: negated variable assignments in if statements by [@kelly-lin](https://github.com/kelly-lin) in [https://github.com/tree-sitter/tree-sitter-bash/pull/183](https://github.com/tree-sitter/tree-sitter-bash/pull/183)
 -   Fixes by [@amaanq](https://github.com/amaanq) in [https://github.com/tree-sitter/tree-sitter-bash/pull/187](https://github.com/tree-sitter/tree-sitter-bash/pull/187)

**Comment by pascalkuthe** [comment]
I am quite aware these is an issue marked as good first issue about it. 

We don't usually work on grammar updates ourself so this just needs some contributor to pick this up

**Comment by David-Else** [comment]
It is being sorted :) https://github.com/helix-editor/helix/pull/10526
